### ISSUE TITLE
Unable to connect standalone Node-RED to CCU2

### ISSUE THREAD

**Issue by m4rc77** [HM-RedMatic/node-red-contrib-ccu#29]
I am currently using hm2mqtt.js. As it's unmaintained i tried to switch to `node-red-contrib-ccu` but with no luck :-(

I installed `node-red-contrib-ccu` on my Node-RED installation which is running on a RaspPi and then tried to connect to my CCU2. I encountered several issues / questions.

- In the `ccu-connection-node` the drop-down for `Listen address` is empty and no value can be entered or selected. Is this normal?
- Is it correct, that both fields `Listen address` and `Init address` can stay empty?
- What values must be used for `BINRPC listening port` and `XMLRPC listening port` or how can i find out what to enter here?

When I then deploy the flow i can see in the logs the following error:
`Nov 11 17:28:33 raspberrypi Node-RED[2173]: 11 Nov 17:28:33 - [error] [ccu-connection:40cce03.b709b2] Error: Bad arguments: First argument should be a string, second should be an array of strings`

Thanks for your help. If things are unclear I can also write in German. 
Greetings from Switzerland.

**Comment by GnomiBerlin** [comment]
Hello m4rc77,
I had also a lot of problems starting with node-red-contrib-ccu. My main problem was the usage of it and Node-Red in a Docker environment. I got some help from hobbyquaker and perhaps my input is helping you.

- CCU address: easy, the CCU IP Address
- Listen address: 172.17.0.x - the internal docker IP address of ...contrib-ccu (was presented as option) / I think it is your RaspPi IP address
- Init address: my server address, where the docker is running / also your RaspPi
- BINRPC and XMLRPC: should differ from each other - if ports are unsed, use 2056 and 2057

I hope this helps.
Greetings from Berlin

**Comment by Hypnos3** [comment]
Have you try to input an IP address in the CCU address field?

**Comment by hobbyquaker** [comment]
Can you please take a look at the Browser Console while opening the ccu-connection Node? I hope there is some error messages that shows us why the dropdown stays empty...

**Comment by hobbyquaker** [comment]
If there is no error message pls take a look at the network tab, there you should see a xhr request to the path `ccu` that should receive a response containing the available listening addresses:

![image](https://user-images.githubusercontent.com/2503307/48638217-41ba5500-e9d0-11e8-8005-97c939f7f480.png)

**Comment by m4rc77** [comment]
I found the following error in the browser console:
`oneditprepare 40cce03.b709b2 ccu-connection TypeError: RED.settings.context is undefined`

I then updated node-red and node-red-contrib-ccu to the latest versions:
- node-red-contrib-ccu: `1.8.3` --> `1.9.1`
- node-red: `0.17.5` --> `0.19.5`

One of those updates did the trick ... i am now able to select the IP of my RasPi as `Listen Address`. I then deployed the flow and the first look is pretty good! As far as I have seen it seems to transfers data from/to CCU.

Thanks a lot for you support!
